AlphaWallet/UI/ButtonsBar.swift

```diff
@@ -3,123 +3,201 @@
 import Foundation
 import UIKit
 
+enum ButtonsBarConfiguration {
+    case empty
+    case combined(buttons: Int)
+    case green(buttons: Int)
+    case white(buttons: Int)
+    case custom(types: [ButtonsBarButtonType])
+    
+    fileprivate static let maxCombinedButtons: Int = 2
+    
+    var buttonViewModels: [ButtonsBarButtonType] {
+        switch self {
+        case .green(let buttons):
+            return (0..<buttons).compactMap { _ in .green }
+        case .white(let buttons):
+            return (0..<buttons).compactMap { _ in .white }
+        case .custom(let types):
+            return types
+        case .combined(let buttons):
+            let buttonsToShow: [ButtonsBarButtonType] = [.green, .white]
+            if buttons > ButtonsBarConfiguration.maxCombinedButtons {
+                return buttonsToShow
+            } else {
+                return [ButtonsBarButtonType](buttonsToShow.suffix(buttons))
+            } 
+        case .empty:
+            return []
+        }
+    } 
+    
+    var showMoreButton: Bool {
+        switch self {
+        case .green, .white:
+            return false
+        case .custom:
+            return false
+        case .combined(let buttons):
+            return buttons >= ButtonsBarConfiguration.maxCombinedButtons
+        case .empty:
+            return false
+        }
+    }
+}
+
+enum ButtonsBarButtonType {
+    case green
+    case white
+    case more
+    
+    var viewModel: ButtonsBarViewModel {
+        switch self {
+        case .green:
+            return ButtonsBarViewModel.greenButton
+        case .white:
+            return ButtonsBarViewModel.whiteButton
+        case .more:
+            return ButtonsBarViewModel.moreButton
+        }
+    }
+}
+
 class ButtonsBar: UIView {
     static let buttonsHeight = CGFloat(48)
     //A gap so it doesn't stick to the bottom of devices without a bottom safe area
     static let marginAtBottomScreen = CGFloat(3)
 
-    private var buttonContainerViews: [ContainerViewWithShadow<UIButton>]
+    private var buttonContainerViews: [ContainerViewWithShadow<UIButton>] = []
+    private var optionsButtonContainerViews: [ContainerViewWithShadow<UIButton>] = []
```

Comment:
- [ ] Can you rename this to `moreButtonContainerViews`? Since we have `showMoreButton` (which is good).
- [ ] Rename `optionsButtonContainerViews` accordingly too